dev/devicelab/lib/framework/runner.dart

```diff
@@ -41,10 +40,16 @@ Future<Map<String, dynamic>> runTask(String taskName, { bool silent = false }) a
     runnerFinished = true;
   });
 
+  final Completer<int> port = new Completer<int>();
+
   final StreamSubscription<String> stdoutSub = runner.stdout
       .transform(const Utf8Decoder())
       .transform(const LineSplitter())
       .listen((String line) {
+    const String portAnnouncement = 'Observatory listening on http://127.0.0.1:';
```

Comment:
This will fail in IPv6

Reply:
(might not be a big deal since we control the devicelab and can just say that it's hard coded to be IPv4....)

Reply:
I've fixed the places that mention 127.0.0.1 explicitly to extract the port in an ipv6-compatible manner but I haven't gone further down the yak shave of making devicelab ipv6-safe.